### PR TITLE
Switch ImageLoader to be a codegen'd turbomodule

### PR DESCRIPTION
**Pull request by acoates-ms**
ImageLoader is currently implemented using a cxxmodule.  This replaces that implementation with a TurboModule that is type checked against the native spec.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8754)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Base/CoreNativeModules.cpp [hunk 1]
```diff
@@ -63,11 +62,6 @@ std::vector<facebook::react::NativeModuleDescription> GetCoreModules(
   modules.emplace_back(
       LinkingManagerModule::name, []() { return std::make_unique<LinkingManagerModule>(); }, batchingUIMessageQueue);
 
-  modules.emplace_back(
-      ImageViewManagerModule::name,
-      []() { return std::make_unique<ImageViewManagerModule>(); },
-      batchingUIMessageQueue);
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/Base/CoreNativeModules.cpp:69]: Looks like this was previously affined to the UI thread. Will moving to JS thread cause issues?

File: vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp [hunk 2]
```diff
@@ -77,96 +58,67 @@ winrt::fire_and_forget GetImageSizeAsync(
...
 
-std::map<std::string, folly::dynamic> ImageViewManagerModule::getConstants() {
-  return {{}};
+void ImageLoader::prefetchImageWithMetadata(
+    std::string uri,
+    std::string queryRootName,
+    double rootTag,
+    React::ReactPromise<React::JSValue> &&result) noexcept {
+  // NYI
+  result.Resolve(true);
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp:114]: I saw that we resolved before, but would rejecting be more appropriate here?

Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp:114]: Same for other functions

File: vnext/Microsoft.ReactNative/Base/CoreNativeModules.cpp (hunk 1 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Base/CoreNativeModules.cpp:69]: TurboModules run on the JS thread, thats why ImageLoader::getSizeWithHeaders posts back to the UI thread.

File: vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp (hunk 2 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp:114]: Possibly, but I dont want to conflate moving this to the more type safe / efficient turbomodule and changing its functionality.
Resolving is certainly safer, as it'll claim that the image is ready to go.  Where apps that are assuming this API is working are much more likely to break if it was rejected.

**Review by NickGerleman — APPROVED** [review]

**Pull request merged**